apps/web/src/pages/settings/user-profile-page/UserProfileForm.tsx

```diff
@@ -57,25 +63,54 @@ export const UserProfileForm: FC<IUserProfileFormProps> = ({ currentUser }) => {
     },
   });
 
-  const onUpdateUserProfileHandler: SubmitHandler<IUserProfileForm> = ({ firstName, lastName }) => {
-    updateUserProfile({ firstName, lastName });
+  const { uploadToStorage, isLoading: isUploadingImage } = useUploadToStorage({
+    onError: (e) => {
+      errorMessage('Failed to upload profile image: ' + e.message);
+    },
+  });
+
+  const uploadProfileImage = async (objectUrl: string): Promise<string> => {
+    const blob = await fetch(objectUrl).then((el) => el.blob());
+    const path = await uploadToStorage({
+      file: new File([blob], 'profile', { type: blob.type }),
+      type: UploadTypesEnum.USER_PROFILE,
+    });
+    URL.revokeObjectURL(objectUrl);
+
+    return path;
   };
 
-  const onUpdateUserProfileImageHandler = async (_: File[]) => {
-    // TODO: Implement image upload
+  const onUpdateUserProfileHandler: SubmitHandler<IUserProfileForm> = async (data) => {
+    const objectUrl = data.profilePicture;
+    let path: string | undefined;
+    if (objectUrl && objectUrl.startsWith('blob:')) {
+      path = await uploadProfileImage(objectUrl);
+    }
```

Comment:
When the user picks the file we convert it to the object URL which is stored in the browser memory. Then that object URL is set on the form so we can quickly show the picked image. During the form submission, we have to upload that file and release it from memory.